Update dnsmasq to noble-updates for USN-7619-1

slyon opened the pull request.

slyon committed:
snapcraft: update dnsmasq to noble-updates

alfonsosanchezbeato commented on .github/workflows/tests.yaml:
```diff
@@ -10,6 +10,9 @@ jobs:
     steps:
+      - name: Prepare
+        run: |
+          mkdir ~/bin
```
Why is this needed now? Maybe it would need a comment.

slyon replied:
This was just a test. I think such change should actually land in https://github.com/canonical/system-snaps-cicd-tools/pull/22 – Can you help with that?


From the previous CI run (https://github.com/canonical/network-manager-snap/actions/runs/16141632221/job/45550249420) I saw this:
```
Artifact network-manager-snaps.zip successfully finalized. Artifact ID 3485352351
Artifact network-manager-snaps has been successfully uploaded! Final size is 14791628 bytes. Artifact ID is 3485352351
Artifact download URL: https://github.com/canonical/network-manager-snap/actions/runs/16141632221/artifacts/3485352351
Run printf "Running spread tests\n"
  printf "Running spread tests\n"
  wget -q https://storage.googleapis.com/snapd-spread-tests/spread/spread-plus-amd64.tar.gz \
      -O - | tar xz
  mv spread ~/bin/spread
  export PATH=$PATH:~/bin/
  # TODO tests for other archs?
  cp "/home/ubuntu/actions-runner/_work/_temp"/network-manager_*_amd64.snap .
  suites=""
  if [ -z "$suites" ]; then
      echo "Using default spread suite"
      spread google:
  else
      spread $suites
  fi
  shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
  env:
    SNAP_NAME: network-manager
    LP_CREDENTIALS: ***
  
Running spread tests
mv: cannot move 'spread' to '/home/ubuntu/bin/spread': No such file or directory
Error: Process completed with exit code 1.
```

slyon committed:
CI: run workflows on 'spread-enabled' runners

This is to gain access to google-cloud credentials and avoid errors like this:
2025-07-08 11:47:27 Cannot allocate google:ubuntu-core-24-64: google: could not find default credentials. See [https://developers.google.com/accounts/docs/application-default-credentials](https://developers.google.com/accounts/docs/application-default-credentials) for more information.
